python/lsst/sims/catUtils/mixins/sncat.py

```diff
@@ -0,0 +1,302 @@
+            sed = SNobject.SNObjectSED(time=self.obs_metadata.mjd,
+                                       bandpass=self.lsstBandpassDict,
+                                       applyExitinction=True)
+            sedlist.append(sed)
+
+        return sedlist
+
+    def get_time(self):
+
+        return np.repeat(self.mjdobs, self.numobjs)
+
+    def get_band(self):
+        bandname = self.obs_metadata.bandpass
+        return np.repeat(bandname, self.numobjs)
+
+    @compound('flux', 'mag', 'flux_err', 'mag_err')
+    def get_snbrightness(self):
+
+        c, x1, x0, t0, _z, ra, dec = self.column_by_name('c'),\
+            self.column_by_name('x1'),\
+            self.column_by_name('x0'),\
+            self.column_by_name('t0'),\
+            self.column_by_name('redshift'),\
+            self.column_by_name('raJ2000'),\
+            self.column_by_name('decJ2000')
+
+        SNobject = SNObject()
+        bandname = self.obs_metadata.bandpass
+        bandpass = self.lsstBandpassDict[bandname]
+
+        # Initialize return array
+        vals = np.zeros(shape=(self.numobjs, 4))
+
+        for i, v in enumerate(vals):
+            SNobject.set(z=_z[i], c=c[i], x1=x1[i], t0=t0[i], x0=x0[i])
+            SNobject.setCoords(ra=ra[i], dec=dec[i])
+            SNobject.mwEBVfromMaps()
+
+            # Calculate fluxes
+            flux = SNobject.catsimBandFluxes(time=self.mjdobs,
+                                             bandpassobject=bandpass)
+            mag = SNobject.catsimBandMags(time=self.mjdobs,
+                                          bandpassobject=bandpass)
+            vals[i, 0] = flux
+            vals[i, 1] = mag
+            flux_err = SNobject.catsimBandFluxError(time=self.mjdobs,
+                                                    bandpassobject=bandpass,
+                                                    m5=self.obs_metadata.m5[
+                                                        bandname],
+                                                    photParams=None,
+                                                    magnitude=mag)
+            mag_err = SNobject.catsimBandMagError(time=self.mjdobs,
+                                                  bandpassobject=bandpass,
+                                                  m5=self.obs_metadata.m5[
+                                                      bandname],
+                                                  photParams=None,
+                                                  magnitude=mag)
+            vals[i, 2] = flux_err
+            vals[i, 3] = mag_err
+        return (vals[:, 0], vals[:, 1], vals[:, 2], vals[:, 3])
+
+    @compound('flux_u', 'flux_g', 'flux_r', 'flux_i', 'flux_z', 'flux_y',
+              'mag_u', 'mag_g', 'mag_r', 'mag_i', 'mag_z', 'mag_y',
+              'adu_u', 'adu_g', 'adu_r', 'adu_i', 'adu_z', 'adu_y', 'mwebv')
+    def get_snfluxes(self):
+
+        c, x1, x0, t0, _z, ra, dec = self.column_by_name('c'),\
```

Comment:
can you have get_snbrightness depend on get_snfluxes (so that code is not duplicated)?